File: android/src/main/java/com/backgroundlocation/event/HttpResponseEvent.java
package com.backgroundlocation.event;

import org.json.JSONObject;

/**
 * HTTP Response Event
 * Type-safe event class
 */
public class HttpResponseEvent {
    private final int status;
    private final boolean success;
    private final String responseText;
    
    public HttpResponseEvent(int status, boolean success, String responseText) {
        this.status = status;
        this.success = success;
        this.responseText = responseText;
    }
    
    public int getStatus() {
        return status;
    }
    
    public boolean isSuccess() {
        return success;
    }
    
    public String getResponseText() {
        return responseText;
    }
    
    public JSONObject toJson() {
        try {
            JSONObject json = new JSONObject();
            json.put("status", status);
            json.put("success", success);
            json.put("responseText", responseText);
            return json;
        } catch (Exception e) {
            return new JSONObject();
        }
    }
    
    public String getEventName() {
        return "http";
    }
}

File: ios/event/HttpResponseEvent.m
//
//  HttpResponseEvent.m
//  RNBackgroundLocation
//
//  HTTP Response Event
//  Android HttpResponseEvent.java benzeri
//

#import "HttpResponseEvent.h"

@implementation HttpResponseEvent

- (instancetype)initWithStatusCode:(NSInteger)status 
                            success:(BOOL)success 
                       responseText:(NSString *)responseText {
    self = [super init];
    if (self) {
        _statusCode = status;
        _status = status;
        _success = success;
        _isSuccess = success;
        _responseText = responseText ?: @"";
        _requestData = nil;
        _responseData = nil;
        _error = nil;
        _locations = nil;
    }
    return self;
}

- (instancetype)initWithStatusCode:(NSInteger)statusCode
                         requestData:(NSDictionary*)requestData
                        responseData:(NSData*)responseData
                               error:(NSError*)error {
    self = [super init];
    if (self) {
        _statusCode = statusCode;
        _status = statusCode;
        _requestData = requestData;
        _responseData = responseData;
        _error = error;
        _success = (statusCode >= 200 && statusCode < 300) && (error == nil);
        _isSuccess = _success;
        
        if (responseData) {
            _responseText = [[NSString alloc] initWithData:responseData encoding:NSUTF8StringEncoding] ?: @"";
        } else {
            _responseText = @"";
        }
        
        _locations = nil;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    return @{
        @"status": @(self.status),
        @"success": @(self.success),
        @"responseText": self.responseText
    };
}

- (NSString *)getEventName {
    return @"http";
}

@end
